src/main/scala/io/iohk/praos/domain/RelativeStakeCalculator.scala

```diff
@@ -29,9 +30,9 @@ object RelativeStakeCalculator {
     *
     * @param publicKey The public key used to identify a particular stakeholder.
     * @param stakeDistribution The stake distribution to be used in the calculations.
-    * @note Precondition: stakeDistribution contains publicKey
     */
   def calculate(publicKey: Key, stakeDistribution: StakeDistribution): RelativeStake = {
+    require(stakeDistribution.contains(publicKey), "No such stakeholder in the stake distribution")
```

Comment:
Maybe we should return that the stakeholder has no stake in that case?
I would prefer avoiding throwing an exception if possible, however this shouldn't be a blocker for merging this PR.

Reply:
Hmm... IMHO we would be hiding a logic error if we return that the stakeholder has no stake in that case.

Reply:
:thinking: I think that depends on the interpretation we give to this function.
We can leave it as is for now but we might probably reconsider it's interpretation when we start using it.